### PR TITLE
one hot encoding

### PR DESCRIPTION
**Pull request by aashokvardhan**

### CODE REVIEW

File: fastestimator/op/numpyop/univariate/onehot.py [hunk 1]
```diff
@@ -55,11 +55,14 @@ def forward(self, data: List[Union[int, np.ndarray]], state: Dict[str, Any]) ->
...
 
     def _apply_onehot(self, data: Union[int, np.ndarray]) -> np.ndarray:
-        class_index = np.array(data)
-        assert "int" in str(class_index.dtype)
-        assert class_index.size == 1, "data must have only one item"
-        class_index = class_index.item()
-        assert class_index < self.num_classes, "label value should be smaller than num_classes"
-        output = np.full((self.num_classes), fill_value=self.label_smoothing / self.num_classes, dtype="float32")
-        output[class_index] = 1.0 - self.label_smoothing + self.label_smoothing / self.num_classes
+        data = np.array(data).astype(np.uint8)
```
Comment (vbvg2008) [inline fastestimator/op/numpyop/univariate/onehot.py:58]: uint8 can represent at most 256 classes.  It won't work on ImageNet (which as 1000 classes)

Comment (vbvg2008) [inline fastestimator/op/numpyop/univariate/onehot.py:58]: we should assert the data that comes in is integer.

File: fastestimator/op/numpyop/univariate/onehot.py [hunk 2]
```diff
@@ -55,11 +55,15 @@ def forward(self, data: List[Union[int, np.ndarray]], state: Dict[str, Any]) ->
...
     def _apply_onehot(self, data: Union[int, np.ndarray]) -> np.ndarray:
-        class_index = np.array(data)
-        assert "int" in str(class_index.dtype)
-        assert class_index.size == 1, "data must have only one item"
-        class_index = class_index.item()
-        assert class_index < self.num_classes, "label value should be smaller than num_classes"
-        output = np.full((self.num_classes), fill_value=self.label_smoothing / self.num_classes, dtype="float32")
-        output[class_index] = 1.0 - self.label_smoothing + self.label_smoothing / self.num_classes
+        data = np.array(data)
+        assert "int" in str(data.dtype)
```
Comment (vbvg2008) [inline fastestimator/op/numpyop/univariate/onehot.py:59]: add an assert message saying the data type must be an integer

**Review by vbvg2008 — COMMENTED** [review]
In the docstring, stating that label starts from 0.

File: fastestimator/op/numpyop/univariate/onehot.py (hunk 2 above)
Comment (aashokvardhan) [reply fastestimator/op/numpyop/univariate/onehot.py]: added assert message

**Review by vbvg2008 — APPROVED** [review]
LG

**Pull request merged**